### PR TITLE
MAGECLOUD-5641: ECE-Tools directory copying is broken on Busybox

### PR DESCRIPTION
**Pull request by shiftedreality**
### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-5568

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Step/Deploy/InstallUpdate/Install/Setup.php [hunk 1]
```diff
@@ -70,11 +79,12 @@ public function execute()
 
             $this->shell->execute('echo \'Installation time: \'$(date) | tee -a ' . $installUpgradeLog);
             $this->shell->execute(sprintf(
-                '/bin/bash -c "set -o pipefail; %s | tee -a %s"',
+                '%s -c "set -o pipefail; %s | tee -a %s"',
+                $this->utilityManager->get(UtilityManager::UTILITY_BASH),
```
Comment (kandy) [inline src/Step/Deploy/InstallUpdate/Install/Setup.php]: Make sense to rename UTILITY_BASH to UTILITY_SHELL

File: src/Util/UpgradeProcess.php [hunk 2]
```diff
@@ -72,7 +72,7 @@ public function execute()
 
         $this->shell->execute('echo \'Updating time: \'$(date) | tee -a ' . $installUpgradeLog);
         $this->shell->execute(sprintf(
-            '/bin/bash -c "set -o pipefail; %s | tee -a %s"',
+            '/bin/sh -c "set -o pipefail; %s | tee -a %s"',
```
Comment (kandy) [inline src/Util/UpgradeProcess.php]: Should we use UtilityManager::UTILITY_BASH here too?

Comment (shiftedreality) [reply src/Util/UpgradeProcess.php]: Added

File: src/Step/Deploy/InstallUpdate/Install/Setup.php (hunk 1 above)
Comment (shiftedreality) [reply src/Step/Deploy/InstallUpdate/Install/Setup.php]: It's `bash` in `which bash` representation. Why would we use `shell`?

Comment (shiftedreality) [reply src/Step/Deploy/InstallUpdate/Install/Setup.php]: Fixed

File: src/Shell/UtilityManager.php [hunk 3]
```diff
@@ -12,8 +12,8 @@
  */
 class UtilityManager
 {
-    const UTILITY_TIMEOUT = 'timeout';
-    const UTILITY_BASH = 'bash';
+    public const UTILITY_TIMEOUT = 'timeout';
+    public const UTILITY_SHELL = 'sh';
```
Comment (bbatsche) [inline src/Shell/UtilityManager.php]: I like this change and agree it's the correct course of action here, but is this not a backwards incompatible change? What version are we planning to release this in?

Comment (shiftedreality) [reply src/Shell/UtilityManager.php]: Good catch - added the previous constant as deprecated

**Review by bbatsche — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

**Comment by shiftedreality** [comment]
Functional changes covered with automated tests.
No QA needed.

**Pull request merged**